src/NServiceBus.Core.Tests/Handlers/MessageHandlerRegistryTests.cs

```diff
@@ -20,6 +22,35 @@ public void ShouldThrowIfUserTriesToBypassTheHandlerContext(Type handlerType)
             Assert.Throws<Exception>(() => registry.RegisterHandler(handlerType));
         }
 
+        [Test]
        public void ShouldIndicateWhetherAHandlerIsATimeoutHandler()
        {
            var registry = new MessageHandlerRegistry(new Conventions());

            registry.RegisterHandler(typeof(SagaWithTimeoutOfMessage));

            var handlers = registry.GetHandlersFor(typeof(MyMessage));
+
```

Comment:
@andreasohlund and I discussed the mix of arrange, act and assert. I proposed to move the asserts together but then we concluded that it is actually better to have the asserts grouped with the timeout handler or the regular handler